src/tools/qt-gui/src/guisettings.cpp

```diff
@@ -113,18 +121,34 @@ void GUISettings::setNodeWithoutKeyColor (const QColor & color)
 	if (color != m_nodeWithoutKeyColor)
 	{
 		m_nodeWithoutKeyColor = color;
-		append (m_nodeWOKeyColorString, color);
+		appendColor (m_nodeWOKeyColorString, color);
 
 		emit nodeWithoutKeyColorChanged ();
 	}
 }
 
-void GUISettings::append (const std::string & keyName, const QColor & color)
+void GUISettings::useSystemIconTheme (const bool & use)
+{
+	if (use != m_useSystemIconTheme)
+	{
+		m_useSystemIconTheme = use;
+		appendBool (m_useSystemIconThemeString, use);
+		emit useSystemIconThemeChanged ();
+	}
+}
+
+void GUISettings::appendColor (const std::string & keyName, const QColor & color)
 {
 	std::string name = "user" + m_base + m_profile + keyName;
 	m_config.append (Key (name, KEY_VALUE, color.name ().toStdString ().c_str (), KEY_END));
 }
 
+void GUISettings::appendBool (const std::string & keyName, const bool value)
+{
+	std::string name = "user" + m_base + m_profile + keyName;
+	m_config.append (Key (name, KEY_VALUE, (value ? "true" : "false"), KEY_END));
```

Comment:
Non-POD types (such as string) and vaargs are a bit dangerous.

What about using `key.set<bool>(value)`?